### DISCUSSION TITLE
Full SPARQL 1.1 feature support for Oxigraph

### DISCUSSION THREAD

**Discussion by cravishe** [oxigraph/oxigraph#412] [General]
I went through the documentation and found that Oxigraph supports [SPARQL 1.1 Query](https://www.w3.org/TR/sparql11-query/), [SPARQL 1.1 Update](https://www.w3.org/TR/sparql11-update/), and [SPARQL 1.1 Federated Query](https://www.w3.org/TR/sparql11-federated-query/) and other features. Does it also support other SPARQL 1.1 features like SPARQL Algebra, Property paths, regular expression etc. In specific, I would like to know if there are any SPARQL 1.1 features oxigraph doesnt support. Any information on this will be helpful.

**Comment by Tpt** [comment]
All of SPARQL 1.1 query and update is supported by Oxigraph with some [details, corner cases and extensions described in this wiki page](https://github.com/oxigraph/oxigraph/wiki/SPARQL). There is also [a crate to access and manipulate SPARQL algebra in Rust](https://crates.io/crates/spargebra).

**Comment by cravishe** [comment]
Thank you for the information. 
Also, I found a list of SPARQL 1.1 test results: https://w3c.github.io/rdf-tests/sparql11/reports/. Are there any such results for SPARQL with Oxigraph?

**Reply by Tpt** [reply]
Oxigraph runs the testsuite as part of its CI ([code](https://github.com/oxigraph/oxigraph/blob/main/testsuite/tests/sparql.rs#L31)).
You see most of the tests pass except some ones. Most of the failures are related to RDF 1.0 vs RDF 1.1 elements and literal normalization.

It also allows to generate implementation report.

For example, running the SPARQL 1.0 tests:
```bash
cd testsuite
 cargo run -- https://w3c.github.io/rdf-tests/sparql/sparql10/manifest.ttl > sparql10-report.ttl
 ```

**Comment by cravishe** [comment]
Thank your for the reply. It's been super helpful
